crates/sui-quorum-driver/src/lib.rs

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use arc_swap::ArcSwap;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::time::Instant;
+
+use tokio::sync::mpsc::{self, Receiver, Sender};
+use tokio::sync::Mutex;
+use tokio::task::JoinHandle;
+use tracing::log::{error, warn};
+use tracing::Instrument;
+
+use sui_core::authority_aggregator::AuthorityAggregator;
+use sui_core::authority_client::AuthorityAPI;
+use sui_types::error::{SuiError, SuiResult};
+use sui_types::messages::{
+    CertifiedTransaction, ExecuteTransactionRequest, ExecuteTransactionRequestType,
+    ExecuteTransactionResponse, Transaction, TransactionEffects,
+};
+
+const DEFAULT_PROCESS_TIMEOUT: Duration = Duration::from_secs(30);
+
+enum QuorumTask<A> {
+    ProcessTransaction(Transaction),
+    ProcessCertificate(CertifiedTransaction),
+    UpdateValidators(AuthorityAggregator<A>),
```

Comment:
Maybe rename UpdateValidators to UpdateCommittee to make it clear you are updating the set of validators rather than performing updates on validators? this confused me at first